lib/samson/logging.rb

```diff
@@ -27,3 +27,23 @@
   config.logger = Syslog::Logger.new('samson')
   config.lograge.formatter = Lograge::Formatters::Logstash.new
 end
+
+# puma kills itself with INT when it receives a SIGHUP see https://github.com/puma/puma/blob/master/docs/signals.md
+# If we have a file based logger then we need to reopen that file to log to the new logfile after log rotation.
+# The rails logger does not know what file it opened, so we have to help it https://github.com/rails/rails/issues/32211
+# If we do not have a file-based logger we still do not want to kill the process, but log that we got the signal.
+# We need to use self-pipe since we cannot reopen logs in an interrupt.
+# This will break stdout_redirect feature of puma, which I hope nobody uses.
+if ENV["SERVER_MODE"]
+  read, write = IO.pipe
+  Signal.trap(:SIGHUP) { write.puts }
+  Thread.new do
+    loop do
+      read.gets
+      Rails.logger.info "Received SIGHUP ... reopening logs"
+      dev = Rails.logger.instance_variable_get(:@logdev)&.dev
+      path = dev.path if dev&.is_a?(File)
+      Rails.logger.reopen(path)
```

Comment:
We want to call `Rails.logger.reopen(nil)` if `dev` isn't a file?

Reply:
Calling with nil is a noop for all the other loggers, so not calling it is fine too ... maybe more obvious what is going on by not calling it